File: iosApp/iosApp/Services/KeyStore/KeyStoreServiceAssembler.swift
// Created by web3d4v on 16/05/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

final class KeyStoreServiceAssembler: AssemblerComponent {
    
    func register(to registry: AssemblerRegistry) {
        
        registry.register(scope: .singleton) { resolver -> KeyStoreService in
            
            DefaultKeyStoreService(
                store: resolver.resolve(),
                keyChainService: resolver.resolve()
            )
        }
    }
}

File: androidApp/src/main/java/com/sonsofcrypto/web3wallet/android/services/KeyStoreServiceAssembler.kt
package com.sonsofcrypto.web3wallet.android.services

import com.sonsofcrypto.web3lib.keyValueStore.KeyValueStore
import com.sonsofcrypto.web3lib.services.keyStore.DefaultKeyStoreService
import com.sonsofcrypto.web3lib.services.keyStore.KeyChainService
import com.sonsofcrypto.web3lib.services.keyStore.KeyStoreService
import com.sonsofcrypto.web3wallet.android.common.AssemblerComponent
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistry
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistryScope
import smartadapter.internal.extension.name

class KeyStoreServiceAssembler: AssemblerComponent {

    override fun register(to: AssemblerRegistry) {

        to.register(KeyStoreService::class.name, AssemblerRegistryScope.SINGLETON) {
            DefaultKeyStoreService(
                it.resolve("${KeyValueStore::class.name}.KeyStoreService"),
                it.resolve(KeyChainService::class.name),
            )
        }
    }
}